### PR TITLE
Add a first version of CHANGELOG

### PR DESCRIPTION
**Pull request by ruippeixotog**
Adds the list of changes for version 0.5.0 and for the next version. A first step for #77.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    `KeyNotFoundException` now, unless the `ConfigConvert` extends the new `AllowMissingKey` trait.
+
+### 0.5.0 (Jan 3, 2017)
+
+- New features
+  - Sealed families are now converted to and from configs unambiguously by using an extra `type` field (customizable) in
+    their config representation; 
+  - New `CoproductHint` trait which allows customization of the derived `ConfigConvert` for sealed families;
+- Breaking changes
+  - The default config representation for sealed families has changed. Configs for such classes that were read
```
Comment (leifwickland) [inline CHANGELOG.md]: Suggestion: Provide a little more guidance on how the new feature works.
```
  - The default config representation for sealed families has changed. 
    - By default Pureconfig now expects to find a `type` field containing the 
      lowercase simple class name of the type to be read. For example for a family
      including `DogConf` and `CatConf`, Pureconfig expects to find a 
      `type: "dogconf"` member in the config file.
    - The old behavior can be restored by putting an implicit instance 
      of `FirstSuccessCoproductHint` in scope (the migration to the new format is strongly recommended though, as the
      previous one may lead to ambiguous behavior).
```

I think that second bullet point is correct. If not, please correct it.

**Review by leifwickland — APPROVED** [review]
Thanks for doing this, @ruippeixotog! 👍

**Review by melrief — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (leifwickland) [inline CHANGELOG.md]: Suggestion: Provide a link to the description of the new behavior in the README.

Comment (melrief) [reply CHANGELOG.md]: I also would like to add the links, people will probably check the CHANGELOG on github and being able to go to the section of the README describing the new feature can be very useful.

Comment (melrief) [reply CHANGELOG.md]: I usually prefer the CHANGELOG to be concise but I can see the importance of describing what's different, in particular with a Breaking changes section. What do you think @ruippeixotog, should we make the CHANGELOG a bit more descriptive about what's going to break? I personally support @leifwickland idea.

Comment (ruippeixotog) [reply CHANGELOG.md]: Yes, I agree with providing more information. In a bigger project there would be a section for logging changes and a separate migration guide; in this case, I don't think there is the need for that.

Comment (ruippeixotog) [reply CHANGELOG.md]: Added @leifwickland suggestion in e5340bd896ccebf090b4099f29be2faf2b77ca26.

Comment (ruippeixotog) [reply CHANGELOG.md]: Good idea. Added in 2420237.

**Comment by melrief** [comment]
Thanks for the great work @ruippeixotog!

**Pull request merged**